python/cdp_missoula_backend/scraper.py

```diff
@@ -8,10 +8,75 @@
 
+# TODO: Handle the sad paths
+def get_scraped_data() -> List:
+    options = webdriver.ChromeOptions()
+    options.add_argument("--ignore-certificate-errors")
+    options.add_argument("--incognito")
+    options.add_argument("--headless")
+    # Chrome Driver v99 - there is a known issue with Chrome v98 and Selenium's is_displayed()
+    driver = webdriver.Chrome("./chromedriver", options=options)
+
+    msla_url = "https://pub-missoula.escribemeetings.com/"
+    driver.get(msla_url)
+
+    # expand the page's list view to render past meetings
+    list_button = driver.find_element(
+        by=By.CLASS_NAME, value="fc-mergedListViewButton-button"
+    )
+    driver.execute_script("arguments[0].click();", list_button)
+    # TODO: Update to wait for the relevant element to be displayed instead
+    time.sleep(3)
+
+    # common ancestor of the anchor that expands the meeting type to display all the meetings,
+    # and the meeting-headers themselves
+    meeting_type_ancestors = driver.find_elements(
+                               by=By.XPATH, value="//div[@class='MeetingTypeList']"
+                             )
+
+    meetings_info = []
+    for meeting_type_ancestor in meeting_type_ancestors:
+        # WARNING: this is very necessary and important - ALL years of meetings are rendered but
+        # the committee titles are hidden, this would be a very expensive line of code to delete
+        if meeting_type_ancestor.is_displayed():
+            # expand this meeting type to render meetings
+            anchor = meeting_type_ancestor.find_element(
+                by=By.XPATH, value=".//a[contains(@class, 'PastMeetingTypesName')]"
+            )
+            driver.execute_script("arguments[0].click();", anchor)
+
+            # wait until the meeting-headers are visible
+            ancestor_id = meeting_type_ancestor.get_attribute("id")
+            WebDriverWait(driver, 10).until(
+                EC.visibility_of_element_located((By.XPATH, f"//div[@id='{ancestor_id}']//div[@class='meeting-header']"))
+            )
+
+            # get all the info we're interested from from each meeting header
+            meetings = meeting_type_ancestor.find_elements(
+                by=By.XPATH, value=".//div[@class='meeting-header']"
+            )
+            for meeting in meetings:
+                if meeting.is_displayed():
+                    _meeting_data = {}
+                    title = meeting.find_element(
+                        by=By.XPATH, value=".//div[@class='meeting-title']"
+                    )
+                    _meeting_data["title"] = title.text
+
+                    date = meeting.find_element(
+                        by=By.XPATH, value=".//div[@class='meeting-date']"
+                    )
+                    _meeting_data["date"] = date.text
+                    meetings_info.append(_meeting_data)
+                
+    return [len(meetings_info), meetings_info]
```

Comment:
TODO: Verify `len(meetings_info)` 10 times and throw an exception if it exceeds some wild number